### ISSUE TITLE
Not Connecting to Moodle Tool

### ISSUE THREAD

**Issue by SolutionsDEV** [celtic-project/wordpress-lti-platform#5]
Hi Stephen,

I am trying to connect to a Moodle Tool which is configured and waiting for the connection from WordPress.
I added some details, but when trying to enable the tool, it shows an error which says "Tools cannot be enabled if they are not fully configured for either LTI 1.0/1.1/1.2 or LTI 1.3, or no private key has been defined."

I expect that the reason is that "Redirection URI" is missing.

In the tool configuration page, it requires many links, some are filled, there is the "Public Keyset URL" which I assume that it is "JWKS URL" while still the redirection is not filled.

In Moodle, the "JWKS URL" is provided as a link, but there is no Public Key, while the "Redirection URI" is not available in Moodle configuration.

In Moodle these are the values that are showing: Tool URL, Initiate login URL, JWKS URL & Deep linking URL.

Would you kindly tell me how to connect the WP Platform with the Moodle Tool?

Thanks a lot,

**Comment by SolutionsDEV** [comment]
Hi again,

An update on this, in the redirect uri, I used the courses page, I am not sure if this is correct as I told before this value is not available in Moodle. I took the "Platform configuration" details provided by the tool setting and entered them manually in the Moodle platform configuration details except the "Access token URL" which is required in Moodle, I entered the platform login page, not sure is this works which is /platform/login.php.

Now the tool is enabled on the platform and platform is showing as active on the Moodle tool.

In content, I could see the shortcode generator, when clicking it, it opens the tool selection, I select the tool and  click on select, a popup window opens, but it is completely white, nothing on it, then when seeing the page source, it is blank.

Any ideas what I may change to get it working.

Thanks again,

**Comment by spvickers** [comment]
Yes, the JWKS URL is the same as the public keyset URL.

If Moodle does not provide a redirection URI, it should be safe to assume that it is the sane as the Tool URL, but you may also need to add the Deep linking URL.  You can check what value is needed by inspecting the message the tool sends in response to an Initiate Login request; this will include a `redirect_uri` parameter.

**Comment by naglfar** [comment]
Hi,
as someone who's been trying to achieve the same thing all morning I got some input on the problem.
I also got as far as embedding a moodle content via shortcode only to land on a blank page (iframe) and while debugging I finally noticed this is the URL path moodle is redirecting me during the authentication:
`/?lti-platform&auth?scope=openid&response_type=id_token&response_mode=form_post&prompt=none&client_id=moodle....`
Notice the question mark after the auth parameter, this seems to be a bug with moodle not properly checking the auth url and blindly adding parameters starting with ?.
For now I patched the wordpress plugin to correctly handle this broken request but ultimately this is something that needs fixing in the moodle LTI plugin.

The next issue I'm facing is that moodle is somehow looking for the resource uuid as 'id' inside the custom parameters block, which seems to stem from the LTI Advantage standards

Edit:
I addressed the issue of the wrong auth url on the official moodle tracker and they created a PR on the upstream version of their fork, so some time in the future this will hopefully be fixed.
